### PR TITLE
feat: Synced New changes from Scaffold Stark to SpeedRun Base Template 

### LINKED ISSUE #171: [CHORE] Sync New changes from Scaffold Stark to SpeedRun Base Template
## Issue Overview

Keep dependencies in-sync between `scaffold-stark` main repo and `speedrunstark`

## Proposed Solutions or Ideas

- Sync fork in a branch 
- Resolve any merge conflicts and dependency errors
- Run unit tests to make sure they work
- Make sure page runs fine

### PR DESCRIPTION
**Pull request by michojekunle**
# Synced New changes from Scaffold Stark to SpeedRun Base Template 

Fixes #171 

## Types of change

- [ ] Feature
- [ ] Bug
- [x] Enhancement

### CODE REVIEW

**Comment by michojekunle** [comment]
Hi @metalboyrick @Nadai2010, this is a pull request regarding the issue I was assigned. 

Please review

File: packages/nextjs/app/debug/_components/contract/TxReceipt.tsx
Comment (Nadai2010) [inline packages/nextjs/app/debug/_components/contract/TxReceipt.tsx:1]: Does not match, use CopyToCliboard and the file as is in main.

https://github.com/Scaffold-Stark/scaffold-stark-2/blob/main/packages/nextjs/app/debug/_components/contract/TxReceipt.tsx

File: packages/nextjs/app/page.tsx
Comment (Nadai2010) [inline packages/nextjs/app/page.tsx:1]: Restore this file, in the base Template only the Example View should appear.

File: packages/nextjs/components/scaffold-stark/Address.tsx [hunk 1]
```diff
@@ -37,6 +36,12 @@ const blockieSizeMap = {
   "3xl": 15,
 };
 
+const TypedCopyToClipboard = CopyToClipboard as unknown as React.FC<{
```
Comment (Nadai2010) [inline packages/nextjs/components/scaffold-stark/Address.tsx:39]: Here I added the TypedCopyClipboard again, remove it, and check that the rest is correct in the file

https://github.com/Scaffold-Stark/scaffold-stark-2/blob/main/packages/nextjs/components/scaffold-stark/Address.tsx

File: packages/nextjs/components/scaffold-stark/ClassHash.tsx [hunk 2]
```diff
@@ -18,6 +18,12 @@ type ClasshashProps = {
   size?: "xs" | "sm" | "base" | "lg" | "xl" | "2xl" | "3xl";
 };
 
+const TypedCopyToClipboard = CopyToClipboard as unknown as React.FC<{
```
Comment (Nadai2010) [inline packages/nextjs/components/scaffold-stark/ClassHash.tsx:21]: Here I added the TypedCopyClipboard again, removed.

File: packages/nextjs/components/scaffold-stark/CustomConnectButton/AddressInfoDropdown.tsx [hunk 3]
```diff
@@ -34,6 +34,12 @@ type AddressInfoDropdownProps = {
   ensAvatar?: string;
 };
 
+const TypedCopyToClipboard = CopyToClipboard as unknown as React.FC<{
```
Comment (Nadai2010) [inline packages/nextjs/components/scaffold-stark/CustomConnectButton/AddressInfoDropdown.tsx:37]: Here I added the TypedCopyClipboard again, removed.

File: packages/nextjs/app/debug/_components/contract/__test__/utilsContract.test.tsx
Comment (Nadai2010) [inline packages/nextjs/app/debug/_components/contract/__test__/utilsContract.test.tsx:1]: No Add test, remove utilsContract.test.tsx and utilsDisplay.test.tsx

File: packages/nextjs/app/debug/_components/contract/__test__/utilsDisplay.test.tsx
Comment (Nadai2010) [inline packages/nextjs/app/debug/_components/contract/__test__/utilsDisplay.test.tsx:1]: No Add test, remove utilsContract.test.tsx and utilsDisplay.test.tsx

File: packages/nextjs/hooks/scaffold-stark/__tests__/useDeployedContractInfo.test.ts
Comment (Nadai2010) [inline packages/nextjs/hooks/scaffold-stark/__tests__/useDeployedContractInfo.test.ts:1]: Hooks Tests should not be part of Speedrun, please mix old and new ones.

Check the ch-0, ch-1, ch-2 and ch-3

File: packages/nextjs/hooks/scaffold-stark/__tests__/useScaffoldContract.test.ts
Comment (Nadai2010) [inline packages/nextjs/hooks/scaffold-stark/__tests__/useScaffoldContract.test.ts:1]: Hooks Tests should not be part of Speedrun, please mix old and new ones.

Check the ch-0, ch-1, ch-2 and ch-3

File: packages/nextjs/public/debug-image.png
Comment (Nadai2010) [inline packages/nextjs/public/debug-image.png:1]: Remove this file

File: packages/nextjs/public/rpc-version.png
Comment (Nadai2010) [inline packages/nextjs/public/rpc-version.png:1]: Remove this file

File: packages/nextjs/public/scaffold-config.png
Comment (Nadai2010) [inline packages/nextjs/public/scaffold-config.png:1]: Remove this file

**Review by Nadai2010 — CHANGES_REQUESTED** [review]

**Comment by michojekunle** [comment]
Alright Thank you so much for the Review @Nadai2010, 

I've implemented the required changes Please review

**Review by Nadai2010 — APPROVED** [review]
For me everything is fine, good job

**Comment by metalboyrick** [comment]
@Nadai2010 for this PR do we want to wait for more changes from scaffold stark? if not i think you can merge it, looks good on my end

**Comment by Nadai2010** [comment]
We can merged, we can't wait for the new changes, at the end of the hack I will do the sync again. Thanks @michojekunle  and @metalboyrick

**Pull request merged**